docs/architecture/module-organization.md

```diff
@@ -0,0 +1,229 @@
+developers to understand the codebase structure and ensure consistent architectural patterns.
+
+## 📂 Module Overview
+
+The modules are organized into several types, each serving a specific purpose in the overall architecture:
+
+```mermaid
+graph TB
+    subgraph APP[App Modules]
+        direction TB
+        APP_TB["`**app-thunderbird**`"]
+        APP_K9["`**app-k9mail**`"]
+    end
+
+    subgraph COMMON[App Common Module]
+        direction TB
+        APP_COMMON["`**app-common**`"]
+    end
+
+    subgraph FEATURE[Feature Modules]
+        direction TB
+        FEATURE_ACCOUNT["`**feature:account**`"]
+        FEATURE_SETTINGS["`**feature:settings**`"]
+        FEATURE_ONBOARDING["`**feature:onboarding**`"]
+        FEATURE_MAIL["`**feature:mail**`"]
+    end
+
+    subgraph CORE[Core Modules]
+        direction TB
+        CORE_UI["`**core:ui**`"]
+        CORE_COMMON["`**core:common**`"]
+        CORE_ANDROID["`**core:android**`"]
+        CORE_NETWORK["`**core:network**`"]
+        CORE_DATABASE["`**core:database**`"]
+        CORE_TESTING["`**core:testing**`"]
+    end
+
+    subgraph LIBRARY[Library Modules]
+        direction TB
+        LIB_AUTH["`**library:auth**`"]
+        LIB_CRYPTO["`**library:crypto**`"]
+        LIB_STORAGE["`**library:storage**`"]
+    end
+
+    subgraph LEGACY[Legacy Modules]
+        direction TB
+        LEGACY_K9["`**legacy**`"]
+        LEGACY_MAIL["`**mail**`"]
+        LEGACY_BACKEND["`**backend**`"]
+    end
+
+    APP ~~~ COMMON
+    COMMON ~~~ FEATURE
+    FEATURE ~~~ CORE
+    CORE ~~~ LIBRARY
+    LIBRARY ~~~ LEGACY
+    
+    APP --> |depends on| COMMON
+    COMMON --> |depends on| FEATURE
+    FEATURE --> |depends on| CORE
+    CORE --> |depends on| LIBRARY
+    COMMON --> |depends on<br>as legacy bridge| LEGACY
+
+    classDef app fill: #d0e0ff, stroke: #0066cc
+    classDef common fill: #d5f5d5, stroke: #00aa00
+    classDef feature fill: #ffe0d0, stroke: #cc6600
+    classDef core fill: #f0d0ff, stroke: #cc00cc
+    classDef library fill: #fff0d0, stroke: #cc9900
+    classDef legacy fill: #f0f0f0, stroke: #999999
+
+    class APP_TB,APP_K9 app
+    class APP_COMMON common
+    class FEATURE_ACCOUNT,FEATURE_SETTINGS,FEATURE_ONBOARDING,FEATURE_MAIL feature
+    class CORE_UI,CORE_COMMON,CORE_ANDROID,CORE_DATABASE,CORE_NETWORK,CORE_TESTING core
+    class LIB_AUTH,LIB_CRYPTO,LIB_STORAGE library
+    class LEGACY_MAIL,LEGACY_BACKEND,LEGACY_K9 legacy
+```
+
+### Module Types
+
+#### 📱 App Modules
+
+The App Modules (`app-thunderbird` and `app-k9mail`) contain the application-specific code, including:
+- Application entry points and initialization logic
+- Final dependency injection setup
+- Navigation configuration
+- Integration with feature modules solely for that application
+- App-specific themes and resources (strings, themes, etc.)
+
+#### 🔄 App Common Module
+
+The `app-common` module acts as the central hub for shared code between both applications. This module serves as the
```

Comment:
This section might benefit from some info on what should and should not go in here, potentially with examples. For a new contributor it might be hard to decide if something is "common" code vs something that needs to be a feature module.